packages/network/test/unit/document_domain_injection_spec.ts

```diff
@@ -0,0 +1,188 @@
+import { expect } from 'chai'
+import { DocumentDomainInjection, OriginBehavior, DocumentDomainBehavior } from '../../lib/document-domain-injection'
+import { URL } from 'url'
+
+describe('DocumentDomainInjection', () => {
+  describe('InjectionBehavior', () => {
+    let injectDocumentDomain: boolean
+    let testingType: 'e2e' | 'component'
+
+    const cfg = () => {
+      return { injectDocumentDomain, testingType }
+    }
+
+    describe('when injectDocumentDomain config is false', () => {
+      beforeEach(() => {
+        injectDocumentDomain = false
+      })
+
+      describe('and testingType is e2e', () => {
+        beforeEach(() => {
+          testingType = 'e2e'
+        })
+
+        it('returns OriginBehavior', () => {
+          expect(DocumentDomainInjection.InjectionBehavior(cfg())).to.be.instanceOf(OriginBehavior)
+        })
+      })
+
+      describe('and testing type is component', () => {
+        beforeEach(() => {
+          testingType = 'component'
+        })
+
+        it('returns OriginBehavior', () => {
+          expect(DocumentDomainInjection.InjectionBehavior(cfg())).to.be.instanceOf(OriginBehavior)
```

Comment:
might be a better test to assert on `shouldInjectDocumentDomain` being `false` over the class instance as `OriginBehavior` could technically change the value to `true` and the test wont catch it

Reply:
The `OriginBehavior` tests would catch it, though.

Reply:
then we should be good. No need for anything actionable here